bbbc021_feature_reduction_tests.py

```diff
@@ -0,0 +1,298 @@
+import pandas as pd
+import os
+import re
+
+from sklearn.decomposition import PCA
+from sklearn.manifold import TSNE
+from sklearn.preprocessing import StandardScaler
+import umap
+
+from microscopium import io
+from microscopium.preprocess import montage_stream
+from microscopium.preprocess import correct_multiimage_illumination
+from microscopium.preprocess import find_background_illumination
+from microscopium.features import default_feature_map
+
+IMAGE_FILE_PATH = "/data/BBBC021/data.broadinstitute.org/bbbc/BBBC021/"
+OUTPUT_FILE_PATH = "/data/bbbc_out/"
+
+FEATURES_FILE = "./all_features.csv"
+DATA_FILE = OUTPUT_FILE_PATH + "Data_scaled.csv"
+DATA_TSNE = OUTPUT_FILE_PATH + "Data_TSNE_scaled.csv"
+DATA_UMAP = OUTPUT_FILE_PATH + "Data_UMAP_scaled.csv"
+
+
+def main():
+    ## get valid filenames and build output filenames
+    filenames, names_illum = get_valid_file_names(IMAGE_FILE_PATH)
+
+    ## run illumination correction on images
+    for i, directory_files in enumerate(filenames):
+        run_illum(directory_files, names_illum[i])
+        print("Directory processed: ", directory_files[0])
+
+    all_names_montage = []
+    ## montage illumed images
+    for i, filename_dir in enumerate(names_illum):
+        sorted_filenames = sorted(filename_dir)
+        names_montage = make_montage_names(sorted_filenames)
+        all_names_montage.extend(names_montage)
+        run_montage(sorted_filenames, names_montage)
+
+    ## run features on images
```

Comment:
Why is this section commented out?

Reply:
Once each section of the code was run, we commented out to speed things up. In a situation where you're running on a raw unprocessed directory, you'd need every part uncommented